cubano-config/src/main/java/org/concordion/cubano/config/ProxyConfig.java

```diff
@@ -21,7 +21,19 @@
     private int proxyPort = 0;
     private String proxyUsername = "";
     private String proxyPassword = "";
-    private String nonProxyHosts;
+    private String nonProxyHosts = "";
+
+    ProxyConfig(PropertyLoader propertyLoader) {
+        proxyIsRequired = propertyLoader.getPropertyAsBoolean("proxy.required", null);
+
+        setProxyFromConfigFile(propertyLoader);
+        setProxyFromSystemProperties();
+        setProxyFromEnvironmentVariables();
+
+        if (proxyIsRequired && proxyHost.isEmpty()) {
+            propertyLoader.getProperty("proxy.host");
```

Comment:
Just wondering what this line is doing? I left it there, but the result isn't being assigned to anything.

Reply:
I've updated code with a comment, all it does is raise an error if the proxy has not been set

Reply:
I think this would be better if we created a new method that was explicit that it was checking rather than returning, eg propertyLoader.checkProperty("proxy.host").

I won't hold up the PR for now, but would like this considered for a future PR.